src/chart_types/partition_chart/layout/utils/calcs.ts

```diff
@@ -49,10 +50,92 @@ export function arrayToLookup(keyFun: Function, array: Array<any>) {
 }
 
 /** @internal */
+export function convertRGBAStringToSeparateValues(rgba: Color) {
+  const [red1, green1, blue1, alpha1 = 1] = rgba
+    .replace(/[^\d.,]/g, '')
+    .split(',')
+    .map((x) => parseFloat(x));
+  return { red: red1, green: green1, blue: blue1, opacity: alpha1 };
+}
+
+/** If the user specifies the background of the container in which the chart will be on, we can use that color
+ * and make sure to provide optimal contrast
+/** @internal */
+export function combineColors(rgba1: Color, rgba2: Color) {
+  // convert parameters to rgba format
+  let rgba1FormattedRGBA = rgba1.startsWith('#') ? RGBATupleToString(HexToRGB(rgba1)) : rgba1;
+  let rgba2FormattedRGBA = rgba2.startsWith('#') ? RGBATupleToString(HexToRGB(rgba2)) : rgba2;
+
+  rgba1FormattedRGBA = /^[a-z]/.test(rgba1) ? RGBATupleToString(chroma(rgba1).rgba()) : rgba1FormattedRGBA;
+  rgba2FormattedRGBA = /^[a-z]/.test(rgba2) ? RGBATupleToString(chroma(rgba2).rgba()) : rgba2FormattedRGBA;
+  const { red: red1, green: green1, blue: blue1, opacity: alpha1 } = convertRGBAStringToSeparateValues(
+    rgba1FormattedRGBA,
+  );
+  const { red: red2, green: green2, blue: blue2, opacity: alpha2 } = convertRGBAStringToSeparateValues(
+    rgba2FormattedRGBA,
+  );
```

Comment:
we can directly use chroma here, something like this should work:
```suggestion
const [red1, green1, blue1, alpha1] = chroma(rgba1).rgba()
const [red2, green2, blue2, alpha2] = chroma(rgba2).rgba()
```

Reply:
Commit f8619e2 for changes! Thanks!